pkg/virt-controller/watch/migration.go

```diff
@@ -791,6 +797,56 @@ func (c *MigrationController) createAttachmentPod(migration *virtv1.VirtualMachi
 	return nil
 }
 
+func (c *MigrationController) handlePendingPodTimeout(migration *virtv1.VirtualMachineInstanceMigration, vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod) error {
+	if pod.Status.Phase != k8sv1.PodPending || pod.DeletionTimestamp != nil {
+		return nil
+	}
+
+	isUnschedulable := false
```

Comment:
I don't think this variable is required, nor the control code to exit the function if the pod is not `k8sv1.PodReasonUnschedulable` .

You could move all the code in https://github.com/kubevirt/kubevirt/pull/6737/files#diff-1baf192bb9c4560d7d40f008b7ba5c680ebce669b8f1b7639445abd3e7269688R820-R844 to a couple of new functions (named `didPodTimeOut(pod *v1.Pod) bool {...}` and `func (c *MigrationController) podTimedOut(vmi, pod) error {...}`) then directly call them when the right condition is met - i.e. `condition.Type == k8sv1.PodScheduled` && `condition.Status == k8sv1.ConditionFalse` && `condition.Reason == k8sv1.PodReasonUnschedulable`.

Not a must though, I won't insist.

Reply:
I refactored this a bit to hopefully make it easier to follow